inspirehep/modules/records/receivers.py

```diff
@@ -344,6 +344,31 @@ def check_if_record_is_going_to_be_deleted(sender, *args, **kwargs):
     If 'deleted' field exists and its value is True, before update,
     then delete all the record's pidstores.
     """
+    def _get_record_collection(ref_obj):
+        if not isinstance(ref_obj, dict):
+            return None
+        url = ref_obj.get('$schema', '')
+        if not url == '':
+            if 'hep' in url:
+                res = 'literature'
+            elif 'authors' in url:
+                res = 'authors'
+            else:
+                res = None
+        else:
+            res = None
```

Comment:
w00t!

I'm sure there is a better way to write this helper. It looks suspiciously close to #1612 and to what #1355 requests, so at least this needs a FIXME comment similar to those I mention there.